### PR TITLE
GUI: Update activation policy for binary blocks

### PR DESCRIPTION
**Pull request by russellhancox**

### CODE REVIEW

**Review by tburgin — APPROVED** [review]

**Review by mlw — APPROVED** [review]

**Review by tburgin — CHANGES_REQUESTED** [review]
Switching the `NSApp.activationPolicy` quickly can result in a leaked icon on the dock...let's think about this some more.

File: Source/gui/SNTNotificationManager.m [hunk 1]
```diff
@@ -125,6 +127,8 @@ - (void)queueMessage:(SNTMessageWindowController *)pendingMsg {
   [self.pendingNotifications addObject:pendingMsg];
 
   if (!self.currentWindowController) {
+    // Add app to Cmd+Tab and Dock.
+    NSApp.activationPolicy = NSApplicationActivationPolicyRegular;
```
Comment (tburgin) [inline Source/gui/SNTNotificationManager.m:131]: I think `queueMessage:` is called from not-the-main-thread. Does NSApp care?

Comment (tburgin) [reply Source/gui/SNTNotificationManager.m:131]: To that point, `self.pendingNotifications` seems to be accessed and written to on different threads. We should also fix that.

Comment (russellhancox) [reply Source/gui/SNTNotificationManager.m]: NSApp doesn't seem to care but the threading issue is a bit surprising. It looks like only queueMessage: is called from a non-main thread so I've moved much of it into a dispatch block.

Comment (tburgin) [reply Source/gui/SNTNotificationManager.m]: Nice.

**Review by tburgin — APPROVED** [review]

**Review by mlw — APPROVED** [review]

**Pull request merged**